datastore/sync_entity_test.go

```diff
@@ -692,6 +727,102 @@ func (suite *SyncEntityTestSuite) TestCreatePBSyncEntity() {
 	suite.Assert().Nil(pbEntity.Specifics)
 }
 
+func (suite *SyncEntityTestSuite) TestDisableSyncChain() {
+	clientID := "client1"
+	id := "disabled_chain"
+	err := suite.dynamo.DisableSyncChain(clientID)
+	suite.Require().NoError(err, "DisableSyncChain should succeed")
+	e, err := datastoretest.ScanTagItems(suite.dynamo)
+	suite.Require().NoError(err, "ScanTagItems should succeed")
+	suite.Assert().Equal(1, len(e))
+	suite.Assert().Equal(clientID, e[0].ClientID)
+	suite.Assert().Equal(id, e[0].ID)
+}
+
+func (suite *SyncEntityTestSuite) TestIsSyncChainDisabled() {
+	clientID := "client1"
+
+	disabled, err := suite.dynamo.IsSyncChainDisabled(clientID)
+	suite.Require().NoError(err, "IsSyncChainDisabled should succeed")
+	suite.Assert().Equal(false, disabled)
+
+	err = suite.dynamo.DisableSyncChain(clientID)
+	suite.Require().NoError(err, "DisableSyncChain should succeed")
+	disabled, err = suite.dynamo.IsSyncChainDisabled(clientID)
+	suite.Require().NoError(err, "IsSyncChainDisabled should succeed")
+	suite.Assert().Equal(true, disabled)
+}
+
+func (suite *SyncEntityTestSuite) TestClearServerData() {
```

Comment:
I wonder why we do separate testing for sync entities and tag items?
Maybe we should check ClearServerData would delete both in one call?

Reply:
`InsertSyncEntitiesWithServerTags` creates both tags and sync entities.  The second call to `ClearServerData` provides a test case that deletes both.